physics/euler_solver_test.cpp

```diff
@@ -535,6 +536,290 @@ TEST_F(EulerSolverTest, ДжанибековEffect) {
+                                             1.0 * SIUnit<AngularMomentum>()}));
+
+    EXPECT_THAT(
+        actual_attitude(e1),
+        Componentwise(AlmostEquals(expected_attitude(e1).coordinates().x, 0, 8),
+                      AlmostEquals(expected_attitude(e1).coordinates().y, 0, 1),
+                      Le(std::numeric_limits<double>::min())));
+    EXPECT_THAT(
+        actual_attitude(e2),
+        Componentwise(AlmostEquals(expected_attitude(e2).coordinates().x, 0, 1),
+                      AlmostEquals(expected_attitude(e2).coordinates().y, 0, 8),
+                      Le(std::numeric_limits<double>::min())));
+    EXPECT_THAT(
+        actual_attitude(e3),
+        Componentwise(Le(std::numeric_limits<double>::min()),
+                      Le(std::numeric_limits<double>::min()),
+                      AlmostEquals(expected_attitude(e3).coordinates().z, 0)));
+  }
+  // A sphere that rotates around an axis with a random orientation.  Also, a
+  // non-trivial initial attitude.
+  {
+    Solver::AngularMomentumBivector const
+        initial_angular_momentum({1.0 * SIUnit<AngularMomentum>(),
+                                  2.0 * SIUnit<AngularMomentum>(),
+                                  4.0 * SIUnit<AngularMomentum>()});
+    Solver::AttitudeRotation const initial_attitude(
+        0.5 * Radian,
+        1.5 * Radian,
+        -2.5 * Radian,
+        EulerAngles::YZY,
+        DefinesFrame<PrincipalAxes>{});
+    Solver const solver2(moments_of_inertia2,
+                         initial_angular_momentum,
+                         initial_attitude,
+                         Instant());
+
+    auto const t = Instant() + 10 * Second;
+    auto const actual_angular_momentum = solver2.AngularMomentumAt(t);
+    auto const actual_angular_velocity =
+        solver2.AngularVelocityFor(actual_angular_momentum);
+    auto const actual_attitude = solver2.AttitudeAt(actual_angular_momentum, t);
+
+    auto const expected_angular_velocity =
+        AngularVelocity<PrincipalAxes>({1.0 / 3.0 * Radian / Second,
+                                        2.0 / 3.0 * Radian / Second,
+                                        4.0 / 3.0 * Radian / Second});
+    auto const expected_angular_frequency = expected_angular_velocity.Norm();
+    auto const expected_attitude =
+        initial_attitude *
+        Rotation<PrincipalAxes, PrincipalAxes>(
+            expected_angular_frequency * 10 * Second, initial_angular_momentum);
+
+    EXPECT_THAT(actual_angular_momentum,
+                AlmostEquals(initial_angular_momentum, 0));
+    EXPECT_THAT(actual_angular_velocity,
+                AlmostEquals(expected_angular_velocity, 0));
+    EXPECT_THAT(actual_attitude(e1), AlmostEquals(expected_attitude(e1), 29));
+    EXPECT_THAT(actual_attitude(e2), AlmostEquals(expected_attitude(e2), 68));
+    EXPECT_THAT(actual_attitude(e3), AlmostEquals(expected_attitude(e3), 6));
+  }
```

Comment:
Add a test with `m` somewhere in the 2-3 plane (in the 2 or 3 direction) for I2=I3.